src/ripple/app/rdb/backend/RelationalDBInterfaceSqlite.cpp

```diff
@@ -603,97 +649,117 @@ RelationalDBInterfaceSqliteImp::deleteAccountTransactionsBeforeLedgerSeq(
 {
-    /* if database exists, use it */
+    if (!useTxTables_)
+        return 0;
+
     if (existsTransaction())
     {
         auto db = checkoutTransaction();
         return ripple::getRows(*db, TableType::AccountTransactions);
     }
 
-    /* else use shard databases */
-    std::size_t rows = 0;
-    iterateTransactionForward(
-        {}, [&](soci::session& session, std::uint32_t shardIndex) {
-            rows += ripple::getRows(session, TableType::AccountTransactions);
-            return true;
-        });
-    return rows;
+    if (shardStoreExists())
+    {
+        std::size_t rows = 0;
+        iterateTransactionForward(
+            {}, [&](soci::session& session, std::uint32_t shardIndex) {
+                rows +=
+                    ripple::getRows(session, TableType::AccountTransactions);
+                return true;
+            });
+        return rows;
+    }
+
+    return 0;
 }
 
 RelationalDBInterface::CountMinMax
 RelationalDBInterfaceSqliteImp::getLedgerCountMinMax()
 {
-    /* if database exists, use it */
     if (existsLedger())
     {
         auto db = checkoutLedger();
         return ripple::getRowsMinMax(*db, TableType::Ledgers);
     }
 
-    /* else use shard databases */
-    CountMinMax res{0, 0, 0};
-    iterateLedgerForward(
-        {}, [&](soci::session& session, std::uint32_t shardIndex) {
-            auto r = ripple::getRowsMinMax(session, TableType::Ledgers);
-            if (r.numberOfRows)
-            {
-                res.numberOfRows += r.numberOfRows;
-                if (res.minLedgerSequence == 0)
-                    res.minLedgerSequence = r.minLedgerSequence;
-                res.maxLedgerSequence = r.maxLedgerSequence;
-            }
-            return true;
-        });
-    return res;
+    if (shardStoreExists())
+    {
+        CountMinMax res{0, 0, 0};
+        iterateLedgerForward(
+            {}, [&](soci::session& session, std::uint32_t shardIndex) {
+                auto r = ripple::getRowsMinMax(session, TableType::Ledgers);
+                if (r.numberOfRows)
+                {
+                    res.numberOfRows += r.numberOfRows;
+                    if (res.minLedgerSequence == 0)
+                        res.minLedgerSequence = r.minLedgerSequence;
+                    res.maxLedgerSequence = r.maxLedgerSequence;
+                }
+                return true;
+            });
+        return res;
+    }
+
+    return {0, 0, 0};
 }
 
 bool
 RelationalDBInterfaceSqliteImp::saveValidatedLedger(
     std::shared_ptr<Ledger const> const& ledger,
     bool current)
 {
-    /* if databases exists, use them */
+    if (!useTxTables_)
+        return true;
+
     if (existsLedger() && existsTransaction())
     {
         if (!ripple::saveValidatedLedger(
                 *lgrdb_, *txdb_, app_, ledger, current))
             return false;
     }
 
-    if (auto shardStore = app_.getShardStore())
+    if (auto shardStore = app_.getShardStore(); shardStore)
```

Comment:
The second condition check is redundant. Should be: '''if (auto shardStore = app_.getShardStore())'''